[WIP] Revise documentation for Swift 5 stdlib additions

natecook1000 opened the pull request:
This is tracking WIP to revise the docs for the additions in [SE-0221](https://github.com/apple/swift-evolution/blob/master/proposals/0221-character-properties.md).

natecook1000 committed:
[stdlib] Revise some Unicode scalar/Character properties

krilnon commented on stdlib/public/core/CharacterProperties.swift:
```diff
@@ -236,25 +238,38 @@ extension Character {
 
-  /// Whether this Character represents a symbol used mathematical formulas
+  /// A Boolean value indicating whether this character represents a symbol
+  /// that naturally appearx in mathematical contexts.
```
```suggestion
  /// that naturally appears in mathematical contexts.
```
x -> s

Dante-Broggi commented on stdlib/public/core/CharacterProperties.swift:
```diff
@@ -96,7 +96,9 @@ extension Character {
   /// nil.
   ///
-  /// Examples:
+  /// A Boolean value indicating whether this character represents a newline.
```
New comment describing newlines, but code is for whole numbers.

natecook1000 committed:
Minor revision to `compactMapValues` docs.

natecook1000 committed:
Continue revisions to Unicode scalar properties docs.

natecook1000 committed:
Continue character and unicode property doc revisions.

natecook1000 commented on stdlib/public/core/CharacterProperties.swift:
```diff
@@ -22,46 +22,47 @@ extension Character {
+  /// - U+000B: LINE TABULATION (VT)
+  /// - U+000C: FORM FEED (FF)
+  /// - "\r" (U+000D): CARRIAGE RETURN (CR)
+  /// - "\r\n" (U+000A U+000D): CR-LF
```
Wow, great catch!

natecook1000 committed:
Add documentation for AdditiveArithmetic, revise Numeric

krilnon committed:
Fix typo: pluralize 'represent'.

krilnon committed:
Wrap ellipses with comments in code examples.

It'll render a bit better this way.

krilnon committed:
Apply minor style updates to count(where:).

- Use the Swift book style for referring to operators by name.
- Pick a more standard introducer for referring to a code example.

natecook1000 committed:
Continue unicode/character property revisions.

krilnon committed:
Revise string interpolation docs.

- Reword some instances of passive voice or unnecessary future tense
- Clarify who's doing what when (you, the type, or the compiler?)
- Standardize on "you" rather than "client"
- Standardize the "for more info, see" blocks
- Fix capitalization of headings
- Prefer contractions
- Remove code voice from a heading
- Reword a mixed prose / code voice hyphenation
- Fix 'pre-allocate' -> 'preallocate'

krilnon committed:
Revise isMultiple(of:) docs.

In particular, add emphasis to mathematical symbols and equations to
match how we document such things elsewhere.

I'm using asterisks for single symbols, and underscores for equations
because it's easier to read in-source when you don't have to escape
multiplication within emphasis.

natecook1000 committed:
Continue revising character properties.

krilnon committed:
Convert table of interpolation examples to a list of examples.

Tables aren't supported by Swift markup, so this wouldn't render
properly in Xcode or on the web.

I also reflowed the paragraph below since I added a few words to avoid
starting a sentence with code voice.

krilnon committed:
Add some abstracts to the SIMD vector types.

- Adds a dictionary of spelled out numbers. Only numbers < 10
  should be spelled out according to editorial.
- Adds abstracts to some of the basic members.

krilnon commented on stdlib/public/core/Integers.swift:
```diff
@@ -105,41 +142,41 @@ public extension AdditiveArithmetic where Self : ExpressibleByIntegerLiteral {
 ///
-/// The following example declares a method that calculates the total of any
-/// sequence with `Numeric` elements.
+/// The following example extends `Seqeunce` with a method that returns an
```
```suggestion
/// The following example extends `Sequence` with a method that returns an
```

Typo on ‘Sequence’.

natecook1000 committed:
Add abstracts for SIMD protocols.

natecook1000 committed:
Add more abstracts for SIMD{n} vector types.

This includes parameter descriptions for the xyzw properties and inits,
but not for the unlabeled initializers. Combined with the protocol
extension method abstracts, this should complete coverage of the concrete
types.

krilnon committed:
Fix `Sequence` typo

Co-Authored-By: natecook1000 <natecook@apple.com>

natecook1000 committed:
Revisions to string interpolation docs.

This makes the description of what a user must implement in a custom
string interpolation type to get the behavior they want.

natecook1000 commented:
These changes are ready for review:

@milseman for the Character/UnicodeScalar properties
@stephentyrone for the SIMD types/protocols
and also @airspeedswift

natecook1000 commented:
@swift-ci Please smoke test

natecook1000 commented on stdlib/public/core/SIMDVectorTypes.swift.gyb:
```diff
@@ -67,6 +85,7 @@ public struct SIMD${n}<Scalar> : SIMD where Scalar: SIMDScalar {
   }
 
 %  for (half,indx) in [('low','i'), ('high',str(n/2)+'+i'), ('even','2*i'), ('odd','2*i+1')]:
+  /// A half-length vector covering the ${half} elements of the vector.
```
@stephentyrone Does this phrasing ("covering the even elements of the vector") sound okay to you? Suggestions?

stephentyrone replied:
"made up of"?

natecook1000 replied:
```suggestion
  /// A half-length vector made up of the ${half} elements of the vector.
```

milseman reviewed (approved):
Character and Scalar properties LGTM

natecook1000 committed:
Wording change for half-width vectors.

natecook1000 commented:
@swift-ci Please smoke test and merge

natecook1000 commented:
@swift-ci Please smoke test
